core/aws-core/src/main/java/software/amazon/awssdk/awscore/auth/AuthSchemePreferenceProvider.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.awscore.auth;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.core.SdkSystemSetting;
+import software.amazon.awssdk.profiles.Profile;
+import software.amazon.awssdk.profiles.ProfileFile;
+import software.amazon.awssdk.profiles.ProfileFileSystemSetting;
+import software.amazon.awssdk.profiles.ProfileProperty;
+import software.amazon.awssdk.utils.Lazy;
+
+@SdkProtectedApi
+public class AuthSchemePreferenceProvider {
+    private final Supplier<ProfileFile> profileFile;
+    private final String profileName;
+
+    private AuthSchemePreferenceProvider(Builder builder) {
+        if (builder.profileFile != null) {
+            this.profileFile = builder.profileFile;
+        } else {
+            this.profileFile = new Lazy<>(ProfileFile::defaultProfileFile)::getValue;
+        }
+
+        if (builder.profileName != null) {
+            this.profileName = builder.profileName;
+        } else {
+            this.profileName = ProfileFileSystemSetting.AWS_PROFILE.getStringValueOrThrow();
+        }
```

Comment:
Same here, use `Validate.getOrDefault`

```suggestion
        this.profileName = Validate.getOrDefault(builder.profileName, 
                                                 ProfileFileSystemSetting.AWS_PROFILE::getStringValueOrThrow);
```